### DISCUSSION TITLE
Pasinobet et mises à jour

### DISCUSSION THREAD

**Discussion by kreon76** [pretrehr/Sports-betting#13] [General]
Bonjour,
Tout d'abord un grand MERCI pour le taf que vous faites sur ce module Python Sports-Betting, c'est un gain de temps incroyable.
J'avais 2 petites choses à vous demander s'il vous plait, j'utilise l'interface graphique et je vois que vous faites des mises à jour de temps en temps (je ne suis pas un pro de Python pardonnez moi). Quelle serait la méthode la + simple pour retrouver vos mises à jour sur mon PC. 
J'ai téléchargé le nouveau dossier sports-betting avec les mises à jour mais ensuite je ne peux plus lancer l'interface j'ai des messages d'erreur car je me doute qu'il faut faire d'autres actions avant de pouvoir profiter du nouveau dossier mis à jour ?

Et 2ème petite chose je suis obligé d'enlever pasinobet car sinon cela plante l'interface, voir message ci-dessous

Merci par avance pour votre retour

Traceback (most recent call last):
  File "C:\temp\Sports-betting-master\interface_pysimplegui.py", line 595, in <module>
    best_match_freebet_interface(window, values)
  File "C:\temp\Sports-betting-master\sportsbetting\interface_functions.py", line 237, in best_match_freebet_interface
    window["ODDS_FREEBET"].update(odds_table(what_was_printed), visible=True)
  File "C:\temp\Sports-betting-master\sportsbetting\interface_functions.py", line 38, in odds_table
    dict_odds = eval("".join(lines[1:i + 1]))
  File "<string>", line 0

SyntaxError: unexpected EOF while parsing

**Comment by pretrehr** [comment]
Bonjour,
Déjà merci pour votre message, je suis content que le module vous aide !

Ensuite, le plus simple pour pouvoir bénéficier des mises à jour facilement est d'installer [Git](https://git-scm.com/downloads) sur votre ordinateur.
Vous pouvez supprimer le projet que vous aviez téléchargé au préalable, on va repartir de zéro 😉

Tout d'abord exécutez cette commande dans un terminal : 
```
git clone https://github.com/pretrehr/Sports-betting
```
Cela va cloner le contenu du projet dans un dossier Sports-betting qui sera créé à l'endroit où vous exécutez la commande. Ce contenu sera alors relié au projet réel (ce qui n'est pas le cas quand on télécharge directement le projet depuis Github). Cette opération est uniquement à réaliser la première fois.

À partir de là, vous pourrez utiliser le projet et lorsque vous voulez obtenir les dernières mises à jour du projet, vous avez simplement à vous placer dans le répertoire "Sports-betting" et effectuer ces commandes successives :
```
git reset --hard
git pull
pip install -r requirements.txt
```


Concernant le problème avec Pasinobet, c'est "normal" que cela ne fonctionne pas. Pasinobet a récemment modifié son site et je n'ai pas encore eu l'occasion d'adapter mon code au nouveau site. Ce sera fait très prochainement. Je vous conseille donc pour l'instant de ne pas faire d'opérations faisant intervenir Pasinobet. 

J'espère que mes explications sont claires. Si vous avez la moindre difficulté ou question, n'hésitez pas à m'en faire part.

**Comment by kreon76** [comment]
Tout d'abord merci pour votre réactivité. et pour pasinobet ce n'est pas bien grave car les cotes ne sont pas toujours très intéressantes donc pas pressé par une future mise à jour ,o))

J'ai installé Git sur mon PC mais je ne suis vraiment pas doué en développement désolé.
J'ai lancé un terminal Git et j'ai cloné en faisant git clone https://github.com/pretrehr/Sports-betting

Quand vous dites d'aller à la racine du projet c'est dans un terminal Git ? et lancer l'interface c'est aussi dans git ?
Car voila les messages que j'ai  quand j'essaie de faire par exemple : 

pip install -r requirements.txt ou python interface_pysimplegui.py

neo@LAPTOP-086GE5S0 MINGW64 /c/python/Sports-betting (master)
$ interface_pysimplegui.py
bash: interface_pysimplegui.py: command not found

neo@LAPTOP-086GE5S0 MINGW64 /c/python/Sports-betting (master)
$ pip install -r requirements.txt
bash: pip: command not found


Est ce un pb de variable python ?
Merci par avance 

Cédric

**Reply by pretrehr** [reply]
Pas de problème !
En fait, il ne faut pas utiliser le terminal Git, les commandes que je vous ai données dans le message précédent peuvent se faire directement depuis le terminal par défaut de votre ordinateur.
Les erreurs que vous obtenez viennent du fait que le terminal Git fonctionne sous Linux donc il y a quelques subtilités supplémentaires. Mais en refaisant les mêmes étapes depuis un terminal Windows, vous ne devriez plus avoir ces erreurs.

**Reply by kreon76** [reply]
Je dois vraiment avoir un problème de variable alors car j'ai ce message dans une cmd windows: 

C:\python\Sports-betting>pip install -r requirements.txt
'pip' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

Du coup j'ai essayé par ce biais : je me suis mis où il y a le program pip
cd C:\Users\neo\AppData\Local\Programs\Python\Python39\Scripts

J'ai lancé donc cette commande : C:\Users\neo\AppData\Local\Programs\Python\Python39\Scripts>pip install -r C:\python\Sports-betting\requirements.txt

et cela me met ces messages : 

Requirement already satisfied: urllib3>=1.25.9 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r C:\python\Sports-betting\requirements.txt (line 1)) (1.26.2)
Requirement already satisfied: selenium>=3.141.0 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r C:\python\Sports-betting\requirements.txt (line 2)) (3.141.0)
Etc....

Pour l'interface c'est identique, quand je passe par une commande Windows cela ne fonctionne pas.

J'ai donc essayé par ce biais : 

C:\Users\neo\AppData\Local\Programs\Python\Python39>python C:\python\Sports-betting\interface_pysimplegui.py

et j'ai ces messages d'erreur : 

C:\Users\neo\AppData\Local\Programs\Python\Python39>python C:\python\Sports-betting\interface_pysimplegui.py
C:\Users\neo\AppData\Local\Programs\Python\Python39\88\chromedriver.exe
None
Traceback (most recent call last):
  File "C:\python\Sports-betting\interface_pysimplegui.py", line 40, in <module>
    sports = get_all_sports()
  File "C:\python\Sports-betting\sportsbetting\database_functions.py", line 716, in get_all_sports
    conn = sqlite3.connect(sb.PATH_DB)
TypeError: expected str, bytes or os.PathLike object, not NoneType


Il y a qqchose que je ne dois pas bien faire j'imagine ?

**Reply by pretrehr** [reply]
Il semble en effet qu'il y a un problème de variable d'environnement.
Sur Windows, allez dans : Panneau de configuration > Système et sécurité > Système > Paramètres système avancés > Variables d'environnement
Dans la partie "Variables utilisateur pour ..." il devrait y avoir une ligne "Path", sélectionnez-là puis "Modifier..." > "Nouveau" puis écrivez le chemin d'accès vers python "C:\Users\neo\AppData\Local\Programs\Python\Python39" et rajoutez également un le chemin d'accès vers pip "C:\Users\neo\AppData\Local\Programs\Python\Python39\Scripts".
Si la variable "Path" n'existe pas, créez-là avec "Nouvelle..." puis rentrez "Path" en nom de la variable et entrez en valeur de la variable "C:\Users\neo\AppData\Local\Programs\Python\Python39; C:\Users\neo\AppData\Local\Programs\Python\Python39\Scripts" (sans les guillemets, mais avec le point-virgule entre les chemins)
OK> OK > ...
Rouvrez un terminal Windows et dans le répertoire du module Sports-betting, réessayez :
`pip install -r requirements.txt`
puis `python interface_pysimplegui.py`

**Reply by kreon76** [reply]
Ah oui avec les variables maintenant je peux lancer dans une cmd Windows mais malheureusement j'ai les mêmes messages qu'auparavant ;o(

c:\python>cd Sports-betting

c:\python\Sports-betting>pip install -r requirements.txt
Requirement already satisfied: urllib3>=1.25.9 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 1)) (1.26.2)
Requirement already satisfied: selenium>=3.141.0 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 2)) (3.141.0)
Requirement already satisfied: Unidecode>=1.1.1 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 3)) (1.1.2)
Requirement already satisfied: numpy>=1.18.4 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 4)) (1.19.5)
Requirement already satisfied: beautifulsoup4>=4.9.1 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 5)) (4.9.3)
Requirement already satisfied: lxml>=4.5.1 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 6)) (4.6.2)
Requirement already satisfied: pysimplegui>=4.29.0 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 7)) (4.34.0)
Requirement already satisfied: fake-useragent>=0.1.11 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 8)) (0.1.11)
Requirement already satisfied: termcolor>=1.1.0 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 9)) (1.1.0)
Requirement already satisfied: chromedriver-autoinstaller>=0.2.2 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 10)) (0.2.2)
Requirement already satisfied: stopit>=1.1.2 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 11)) (1.1.2)
Requirement already satisfied: colorama>=0.4.1 in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 12)) (0.4.4)
Requirement already satisfied: soupsieve>1.2; python_version >= "3.0" in c:\users\neo\appdata\local\programs\python\python39\lib\site-packages (from beautifulsoup4>=4.9.1->-r requirements.txt (line 5)) (2.1)
WARNING: You are using pip version 20.2.3; however, version 21.0 is available.
You should consider upgrading via the 'c:\users\neo\appdata\local\programs\python\python39\python.exe -m pip install --upgrade pip' command.

Mais peut être que ce n'est pas grave car j'arrive bien à lancer la console ?

Une dernière question c'est promis ;o) Quand je lance l'interface pour le tennis j'ai des messages (cf PJ) me demandant "créer une nouvelle donnée sur...'
Est ce normal ? et si oui dois je cliquer sur oui ou non ?



![python_2021-01-31_18-51-18](https://user-images.githubusercontent.com/78261069/106393212-25126800-63f6-11eb-8863-550375747195.png)

**Reply by pretrehr** [reply]
Les messages que vous obtenez dans la console ne sont pas des messages d'erreur, ils sont juste là à titre informatif, vous pouvez les ignorer.

La fenêtre "Créer une nouvelle donnée pour ..." indique simplement qu'une équipe ou joueur n'est pas encore dans la base de données. J'essaie de la maintenir à jour le plus possible mais il se peut qu'il y ait parfois des données manquantes (comme ici).
Ici, vous pouvez cliquer sur "Yes" pour ajouter le joueur à la base de données et vous n'aurez plus le message d'erreur.

**Reply by kreon76** [reply]
Un grand MERCI encore !!! Si vous avez besoin d'un beta testeur n'hésitez pas je serai ravi d'aider si vous avez besoin

**Reply by pretrehr** [reply]
Merci à vous, je n'y manquerai pas. Et n'hésitez pas si vous avez le moindre souci !
(J'ai modifié un peu le code pour que vous n'ayez plus jamais de fenêtre "Créer une nouvelle donnée pour...", parce qu'avec notamment l'Open d'Australie qui arrive, il risque d'y avoir pas mal de messages de la sorte et ça peut rapidement devenir envahissant.)

**Reply by kreon76** [reply]
Bon ben ça marche nickel !! C'est fou la puissance de l'outil.
Je n'y connais pas grand chose mais je vais essayer de voir si j'arrive avec vos résultats à faire ressortir les TRJ des matchs avec les meilleures cotes des Books (cela peut aider pour d"eventuels surebets). Bon ok ce n'est pas gagné avec mes connaissances en dev mais on ne sait jamais ;o))
